### PR TITLE
Make GPT Crawler a CLI

### LINKED ISSUE #37: Are you interested in packaging it as a CLI?
Hey, all! Awesome project! I loved it, it's very useful. 

Do you have any interest in packaging it as a CLI as well? It occurs to me that it would be more straightforward/developer-friendlky to use it this way:

```sh
gpt-crawler --url https://www.builder.io/c/docs/developers --match https://www.builder.io/c/docs/** --selector .docs-builder-container --maxPagesToCrawl 50 --outputFileName output.json
```

Let me know. I would love to discuss this contribution :wink: 

> I got a working example on my fork: https://github.com/marcelovicentegc/gpt-crawler/blob/main/src/cli.ts

### PR DESCRIPTION
**Pull request by marcelovicentegc**
> 1k+ changes from package-lock.json

#### Essentially:
- Makes GPT Crawler a CLI.
- Splits the code between:
     - `core.ts`: the core methods live here
     - `main.ts` leverages the methods from `core.ts` to run the service with config file
     - `cli.ts`: leverages the methods from `core.ts` to run the service as a CLI

#### What are the next steps? 

- [ ] To setup this package under @BuilderIO on NPM (or any other registry). We would basically just need a NPM_TOKEN with write access setup on this repository. Can you help @steve8708 :hugs:?
- [ ] Are you ok on using GItHub workflows to ship the CLI?
- [ ] Are you ok on using semantic-release to take care of versioning this package?

I can address all these on different PRs but need your feedback. Let's keep in touch :wink: .

#### Other changes that are not directly related to the CLI and that I can revert back to what it was or open other PRs:
- Adds `preinstall` script for automatic playwright setup 
- Improves the `Config` interface jsDocs with `@example` and `@default` decorators
- Adds a table of contents to the docs and separates CLI concerning content from development content

Resolves #37

### CODE REVIEW

**Comment by databill86** [comment]
Thank you for this pull request. 
I was wondering if it's possible to expose the service as an API. It would be a lot easier and simpler to run it locally, without the need to publish the gpt crawler. I'm no expert in js, I tried to implement an express js server with the help of chatgpt, but I had a lot of exceptions and errors, so I gave up ^^

**Comment by steve8708** [comment]
This is awesome @marcelovicentegc! Also just realized that you work at VTEX, awesome, I know Mariano and Geraldo (they are amazing) and I went to VTEX Day last year!

To answer your questions: lets definitely setup the workflow process through github actions. I can happily add an NPM_TOKEN to the secrets there. And semantic release is perfect.

Also, I recall you saying this was WIP, is it ready for review on your side or worth still waiting on?

**Comment by marcelovicentegc** [comment]
How cool, @steve8708! This is awesome! I was at VTEX Day this year ❤️. It's such a great event. 

I will set up semantic-release + GitHub actions on another PR, since this one is already large, ok? 🤗 

And finally, the PR is rebased and good for review 👍

File: README.md [hunk 1]
```diff
@@ -1,51 +1,63 @@
...
+#### Install dependencies
 
 ```sh
 npm i
 ```
 
-If you do not have Playwright installed:
-```sh
-npx playwright install
-```
```
[the remark this answers is not in the document]
Comment (steve8708) [reply README.md:39]: unless perhaps there is another solution to this (e.g. a postinstall step)

thinking through the CLI use case, it certainly would be ideal to not have to run this manually 🤔

[the remark this answers is not in the document]
Comment (marcelovicentegc) [reply README.md:39]: I have put this in the `preinstall` step, so people don't have to manually run this command when running gpt-crawler with the config file. It works well as well when developing the CLI as well. 

But once it is shipped we should expect playwright to be installed and configured by the CLI itself, don't you think? We can even prompt the user if they want the CLI to auto install it or if they want to install it themselves.

File: README.md [hunk 2]
```diff
@@ -69,23 +81,51 @@ type Config = {
...
+#### [Running in a container with Docker](./containerapp/README.md)
+
+To obtain the `output.json` with a containerized execution. Go into the `containerapp` directory. Modify the `config.ts` same as above, the `output.json`file should be generated in the data folder. Note : the `outputFileName` property in the `config.ts` file in containerapp folder is configured to work with the container.
+
+#### Running as a CLI
+
+<!-- TODO: Needs to be actually published -->
+
+```sh
+npm i -g @builder.io/gpt-crawler
```
[the remark this answers is not in the document]
Comment (marcelovicentegc) [reply README.md:112]: Yes! Absolutely!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (steve8708) [reply README.md:39]: ah! I missed that, preinstall works great

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (steve8708) [reply README.md:112]: great, if you want to commit that as an update this should be good to merge!

[the remark this answers is not in the document]
Comment (marcelovicentegc) [reply README.md]: Done 😉

**Pull request merged**